pkg/formatter/formatter.go

```diff
@@ -135,3 +135,16 @@ func FormatPorts(labelMap map[string]string) string {
 func TimeSinceInHuman(since time.Time) string {
 	return fmt.Sprintf("%s ago", units.HumanDuration(time.Since(since)))
 }
+
+func FormatLabels(labelMap map[string]string) string {
+	strs := make([]string, 0, len(labelMap))
+	for i := range labelMap {
+		// exclude labels of nerdctl
```

Comment:
Why?

Reply:
When a container has multiple labels, there is no constraint of labels' order.
So, the test should check labels using map.
And, current version cannot guarantee to display only user defined labels. 
For example, my examples display`io.containerd.image.config.stop-signal=SIGTERM` when the containers are terminated.
In other words, the result of ps command can contains internal labels to control them internally.

Reply:
I'm not sure we want to hide nerdctl labels

Reply:
Me, too.
But, I don't think internal important info are exposed throught labels (like `namespace`, `network` and `ports`).
I understood that `Prefix` exists for it.
Is it the right way to expose them?

Reply:
I'll try to find other examples and docker running way.

Reply:
In [docker cli](https://github.com/docker/cli/blob/9cda6ce0114b64052eb656b4975ab03acb8cb371/cli/command/formatter/container.go#L239), it doesn't filter any labels.
But, the reason why it can do is that [moby](https://github.com/moby/moby/blob/98d8343aa28de0d499464b5529e6b8ccc92e9313/api/types/types.go#L152) containers have their various properties (like `Ports` and `NetworkSettings`).
nerdctl cannot do like it. Because it internally saves these kinds of values into Labels(like [this](https://github.com/containerd/nerdctl/blob/b5cdd8aa43c6b40502fc740a14682873574470f9/cmd/nerdctl/ps.go#L216)).
For these implementations, `Prefix` is defined (we can see from [this commit](https://github.com/containerd/nerdctl/commit/7cccafe5c2f684a959f8f0ee7a9e3b014d239235)).
Eventually, I guess that `Prefix` can be understood for only internal implementation.
Users should not know how to implement functions (like ports, networks and so on).

My suggestions are
- Hide labels with `Prefix`
- Explicitly comment "`Prefix` is for internal labels so they will be hide"

Reply:
I really agree we are not sure which is correct.

In [docker cli](https://github.com/docker/cli/blob/9cda6ce0114b64052eb656b4975ab03acb8cb371/cli/command/formatter/container.go#L239), it doesn't filter any labels.
But, the reason why it can do is that [moby](https://github.com/moby/moby/blob/98d8343aa28de0d499464b5529e6b8ccc92e9313/api/types/types.go#L152) containers have their various properties (like `Ports` and `NetworkSettings`).
nerdctl cannot do like it. Because it internally saves these kinds of values into Labels(like [this](https://github.com/containerd/nerdctl/blob/b5cdd8aa43c6b40502fc740a14682873574470f9/cmd/nerdctl/ps.go#L216)).
For these implementations, `Prefix` is defined (we can see from [this commit](https://github.com/containerd/nerdctl/commit/7cccafe5c2f684a959f8f0ee7a9e3b014d239235)).
Eventually, I guess that `Prefix` can be understood for only internal implementation.
Users should not know how to implement functions (like ports, networks and so on).

---
In my opinion, if we don't hide any label, it can be abused to know internal information.
My suggestions are
- Hide labels with `Prefix`
- Explicitly comment "`Prefix` is for internal labels so they will be hide"

Reply:
I agree with @minuk-dev 

The final containerd containers created by nerdctl have some labels because the nerdctl needs those labels to interact with the OCI hook. That means that those labels represent some internal status. I think it would not be necessary to explore those labels for the user, and the user shouldn't care about how those labels work.

Reply:
For a personal suggestion: maybe we can let people decide what they want to do? Here are the details 

1. The default behavior is that we will filter the internal labels

2. Add an argument --all-labels to allow the user to use this to display all the labels, including the internal labels

Reply:
Hiding internal labels can be discussed in a separate issue/ PR.

Having --all-labels is probably fine but it has to be consistent with other "inspect" comnands

Reply:
Okay, should I remove filtering internal label and push again?

Reply:
I think we can hide labels by default, and print when `--labels`.

For internal labels, we can keep them for now.
And another follow-up PR to change the behavior for all the `label` related.

also discussed in this PR:
https://github.com/containerd/nerdctl/pull/1265#discussion_r930080531

Reply:
> Okay, should I remove filtering internal label and push again? 

Yes please 

> or just close this pr?

No